libs/base/Language/Reflection/TTImp.idr

```diff
@@ -209,22 +248,34 @@ mutual
                    (opts : List FnOpt) ->
                    (type : ITy) ->
                    IClaimData
-
+  ||| A top-level declaration
   public export
   data Decl : Type where
-       IClaim : WithFC IClaimData -> Decl
-       IData : FC -> WithDefault Visibility Private -> Maybe TotalReq -> Data -> Decl
-       IDef : FC -> Name -> (cls : List Clause) -> Decl
+       ||| A type ascription, `a : b`.
```

Comment:
Type ascription term is used then you try to refine the type of an expression. I think, this should be called "a function signature"